### PR TITLE
reading gzipped csv files

### PR DESCRIPTION
**Pull request by dulinda**
## What do these changes do?

pd.read_csv is able to open gzipped files 

## Issue Number

Resolves .gz files [#630](https://github.com/modin-project/modin/issues/630)

### CODE REVIEW

File: modin/engines/ray/generic/io.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def file_exists(file_path):
     return os.path.exists(file_path)
 
 
-def file_open(file_path, mode="rb"):
+def file_open(file_path, mode="rb", kwargs=None):
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
def file_open(file_path, mode="rb", compression="infer"):
```

File: modin/engines/ray/generic/io.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ def file_open(file_path, mode="rb"):
             except NoCredentialsError:
                 s3fs = S3FS.S3FileSystem(anon=True)
                 return s3fs.open(file_path)
+        elif "compression" in kwargs:
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
        elif compression == "gzip":
```

File: modin/engines/ray/generic/io.py [hunk 3]
```diff
@@ -49,6 +49,10 @@ def file_open(file_path, mode="rb"):
...
+            if kwargs["compression"] == "gzip":
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
```

File: modin/engines/ray/generic/io.py [hunk 4]
```diff
@@ -49,6 +49,10 @@ def file_open(file_path, mode="rb"):
...
+                import gzip
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
            import gzip
```

File: modin/engines/ray/generic/io.py [hunk 5]
```diff
@@ -49,6 +49,10 @@ def file_open(file_path, mode="rb"):
...
+                return gzip.open(file_path, mode=mode)
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
            return gzip.open(file_path, mode=mode)
```

File: modin/engines/ray/generic/io.py [hunk 6]
```diff
@@ -303,13 +307,19 @@ def _read_csv_from_file_pandas_on_ray(cls, filepath, kwargs={}):
             # step has to happen without removing the `index_col` otherwise it will not
             # be assigned correctly
             kwargs["index_col"] = None
+            f = file_open(filepath, "rb", kwargs)
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
            f = file_open(filepath, "rb", kwargs.get("compression", "infer"))
```

File: modin/engines/ray/generic/io.py [hunk 7]
```diff
@@ -303,13 +307,19 @@ def _read_csv_from_file_pandas_on_ray(cls, filepath, kwargs={}):
...
+            kwargs_uncompressed = kwargs.copy()
+            kwargs_uncompressed["compression"] = None
             names = pandas.read_csv(
-                file_open(filepath, "rb"), **dict(kwargs, nrows=0, skipfooter=0)
+                f, **dict(kwargs_uncompressed, nrows=0, skipfooter=0)
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]: Can you add some inline comments about why the compression must be `None` for the single line case?

File: modin/engines/ray/generic/io.py [hunk 8]
```diff
@@ -325,7 +335,7 @@ def _read_csv_from_file_pandas_on_ray(cls, filepath, kwargs={}):
             skiprows=None,
             parse_dates=parse_dates,
         )
-        with file_open(filepath, "rb") as f:
+        with file_open(filepath, "rb", kwargs) as f:
```
Comment (devin-petersohn) [inline modin/engines/ray/generic/io.py]:
```suggestion
        with file_open(filepath, "rb", kwargs.get("compression", "infer")) as f:
```

File: modin/engines/ray/pandas_on_ray/io.py [hunk 9]
```diff
@@ -80,10 +80,11 @@ def _read_csv_with_offset_pandas_on_ray(
             default Index.
     """
     index_col = kwargs.get("index_col", None)
-    bio = file_open(fname, "rb")
+    bio = file_open(fname, "rb", kwargs)
     bio.seek(start)
     to_read = header + bio.read(end - start)
     bio.close()
+    kwargs["compression"] = 'infer'
```
Comment (devin-petersohn) [inline modin/engines/ray/pandas_on_ray/io.py]: Are we passing in the `compression` in `kwargs`?

File: modin/pandas/test/test_io.py [hunk 10]
```diff
@@ -375,6 +375,18 @@ def teardown_fwf_file():
         os.remove(TEST_FWF_FILENAME)
 
 
+def test_read_csv_gzip():
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/test/test_io.py]: I agree, but I think we can fix this in a future PR.

**Comment by williamma12** [comment]
Sounds good!

**Comment by binary-signal** [comment]
guys do you need any help for this PR ? Reading compressed file for me sounds like a great idea.
Feel free to add me in the loop

Cheers :)

**Pull request merged**